atomate/utils/database.py

```diff
@@ -118,18 +157,68 @@ def from_db_file(cls, db_file, admin=True):
                              "credentials are available.")
 
         if admin:
-            user = creds.get("admin_user")
-            password = creds.get("admin_password")
+            user = creds.get("admin_user", "")
+            password = creds.get("admin_password", "")
         else:
-            user = creds.get("readonly_user")
-            password = creds.get("readonly_password")
+            user = creds.get("readonly_user", "")
+            password = creds.get("readonly_password", "")
 
+        maggma_kwargs = creds.get("maggma_store", {})
         kwargs = creds.get("mongoclient_kwargs", {})  # any other MongoClient kwargs can go here ...
 
         if "authsource" in creds:
             kwargs["authsource"] = creds["authsource"]
         else:
             kwargs["authsource"] = creds["database"]
 
-        return cls(creds["host"], int(creds.get("port", 27017)), creds["database"], creds["collection"],
-                   user, password, **kwargs)
+        return cls(host=creds["host"],
+                   port=int(creds.get("port", 27017)),
+                   database=creds["database"],
+                   collection=creds["collection"],
+                   user=user,
+                   password=password,
+                   maggma_store_kwargs=maggma_kwargs, **kwargs)
+
+    def get_store(self, store_name : str):
+        """Get return the defaultdict if it exists, if not create and return it.
+
+        Args:
+            store_name : name of the store desired
+        """
+        if store_name not in self._maggma_stores:
+            if self._maggma_store_type == 's3':
+                 self._maggma_stores[store_name] = self._get_s3_store(store_name)
+            # Additional stores can be implemented here
+            else:
+                raise NotImplementedError("Maggma store type not currently supported.")
+        return self._maggma_stores[store_name]
```

Comment:
Could you make this return None if self._maggma_store_type is None and maybe issue a warning. Just to avoid confusion.

Reply:
Done